src/opover.c

```diff
@@ -1313,7 +1313,7 @@ Expression *compare_overload(BinExp *e, Scope *sc, Identifier *id)
         if (s_r)
         {
             functionResolve(&m, s_r, e->loc, sc, tiargs, e->e2->type, &args1);
-            if (m.lastf && m.lastf->errors)
+            if (m.lastf && (m.lastf->errors || m.lastf->semantic3Errors))
```

Comment:
Any idea why a separate semantic3Errors field was introduced in 767708299a408b4a234ee9ad4f45b0e6e0c21e6d instead of setting errors?